dev/archery/archery/cli.py

```diff
@@ -550,19 +649,34 @@ def benchmark_diff(ctx, src, preserve, output, cmake_extras,
     # This should not recompute the benchmark from run.json
     archery --quiet benchmark diff WORKSPACE run.json > result.json
     """
+    if language == "cpp":
+        ctx.forward(benchmark_diff_cpp, **kwargs)
+    elif language == "java":
+        ctx.forward(benchmark_diff_java, **kwargs)
+
+
+@benchmark.command(name="diff_cpp")
+@click.pass_context
+def benchmark_diff_cpp(ctx, src, preserve, output, language, cmake_extras,
```

Comment:
Again, why separate the languages' benchmarks? It seems especially valuable to have a clearly unified strategy for comparison of benchmarks

Reply:
This separation comes from `@click.pass_context`. To use different options between two languages (e.g. `java-home` or `cxx-flags`), my understanding was that we need different methods with the `@click.pass_context`.

Let me try to have one method with @click.pass_context by lazily parse arguments for each language.